tools/flang1/flang1exe/semfunc.c

```diff
@@ -4264,11 +4317,26 @@ ref_intrin(SST *stktop, ITEM *list)
       } else if (argdtype == DT_WORD) {
       }
     }
+
+    if (intrin == I_MAX || intrin == I_MIN) {
+      check_max_min_argument(argdtype, &dtype_new, &dtype_last);
+    }
+
     if (!dtype1) {
       f_dt = dtype1 = argdtype; /* Save 1st arg's data type */
       if (DTY(argdtype) == TY_ARRAY)
         break;
     } else {
+      if (intrin == I_MAX || intrin == I_MIN) {
+        int dtypecompare = DTYG(argdtype);
+        if (dtypecompare == TY_WORD) dtypecompare = TY_INT;
+        if (dtypecompare == TY_DWORD) dtypecompare = TY_INT8;
+        if (dtypecompare > maxtype) {
+          maxtype = dtypecompare;
+          dtype1 = argdtype;
```

Comment:
Can you clarify why the argument type checks and promotion cannot be done for `I_MAX` and `I_MIN` in the `intrinsic_error` block? It semes that that block of code was designed to handle these two intrinsics.

Reply:
The logic in this function is relatively complex.

For `intrinsic_error` ,these situations will occur:

The first scenario is as follows：
```
program test
  implicit none
  integer :: a, b, c
  a = 1
  b = 2
  c = max(a)
end program test
```
The second scenario is as follows：
```
program test
  implicit none
  integer :: a, b, c
  a = 1
  b = 2
  c = max()
end program test
```
I understand that when there is an obvious error in the arguments, there are no arguments or the arguments number is wrong, in this case `intrinsic_error` will be called.

And For logic in the `intrinsic_error` block, it is prepared for generic situations：
As follows:
```
/* Need to add a check for min and max first */
if (STYPEG(sptre) == ST_GENERIC && (intrin == I_MAX || intrin == I_MIN)) {
```
So I don’t dare to modify this logic.

And For our case, for example, argument type program will not trigger `intrinsic_error`.
For example:
```
program test
  implicit none
  integer :: a, c
  real :: b
  a = 1
  b = 2
  c = max(a, b)
end program test
```

In this case, the compilation can still pass, but the native version of the compiler will occur a warning.
```
F90-W-0093-Type conversion of expression performed
```
This is what a general check of this logic reports，`chktyp(sp, argtyp, TRUE);` in this logic will be checked, and this check is for many kinds of intrinsics.
```
             if (((intrin == I_BGE) || (intrin == I_BGT) ||
                  (intrin == I_BLE) || (intrin == I_BLT)) &&
                 (SST_IDG(sp) == S_CONST || SST_IDG(sp) == S_IDENT)) {
               mkexpr1(sp);
             } else
               chktyp(sp, argtyp, TRUE);
```

So I haven’t found a perfect method yet. For type judgment and type selection, I chose to add judgment and type update logic in front like this PR.